packages/origin-backend/src/pods/admin/admin.controller.ts

```diff
@@ -0,0 +1,35 @@
+import { IUser, SupportedEvents, UserStatusChangedEvent } from '@energyweb/origin-backend-core';
+import { Body, Controller, Get, Param, Put, UseGuards } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { NotificationService } from '../notification/notification.service';
+import { AdminService } from './admin.service';
+
+@Controller('admin')
+export class AdminController {
+    constructor(
+        private readonly notificationService: NotificationService,
+        private readonly adminService: AdminService
+    ) {}
+
+    @Get('users')
+    @UseGuards(AuthGuard('jwt'))
+    public async getAllUsers() {
+        return this.adminService.getAllUsers();
```

Comment:
Since we know that this endpoint can be only used by Admin and Agent (SupportAgent) roles, please add the authorization.

An example snippet:

```
 @Get('/:id/devices')
    @UseGuards(AuthGuard('jwt'))
    async getDevices(@Param('id') id: string, @UserDecorator() loggedUser: IUserWithRelationsIds) {
        if (!isRole(loggedUser, Role.DeviceManager)) {
            throw new ForbiddenException();
        }
```